src/examples/java/com/amazonaws/crypto/examples/DataKeyCaching/cloudformation-template.yaml

```diff
@@ -0,0 +1,221 @@
+        Type: AWS::Lambda::EventSourceMapping
+        Properties:
+            BatchSize: 1
+            Enabled: true
+            EventSourceArn: !Sub arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:stream/${InputStream}
+            FunctionName: !Ref PythonLambdaFunction
+            StartingPosition: TRIM_HORIZON
+    JavaLambdaOutputTable:
+        Type: AWS::DynamoDB::Table
+        Properties:
+            AttributeDefinitions:
+                -
+                    AttributeName: id
+                    AttributeType: S
+            KeySchema:
+                -
+                    AttributeName: id
+                    KeyType: HASH
+            ProvisionedThroughput:
+                ReadCapacityUnits: 1
+                WriteCapacityUnits: 1
+    JavaLambdaRole:
+        Type: AWS::IAM::Role
+        Properties:
+            AssumeRolePolicyDocument:
+                Version: 2012-10-17
+                Statement:
+                    -
+                        Effect: Allow
+                        Principal:
+                            Service: lambda.amazonaws.com
+                        Action: sts:AssumeRole
+            ManagedPolicyArns:
+                - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+            Policies:
+                -
+                    PolicyName: JavaLambdaAccess
+                    PolicyDocument:
+                        Version: 2012-10-17
+                        Statement:
+                            -
+                                Effect: Allow
+                                Action:
+                                    - dynamodb:DescribeTable
+                                    - dynamodb:BatchWriteItem
+                                Resource: !Sub arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${JavaLambdaOutputTable}
+                            -
+                                Effect: Allow
+                                Action:
+                                    - dynamodb:PutItem
+                                Resource: !Sub arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${JavaLambdaOutputTable}*
+                            -
+                                Effect: Allow
+                                Action:
+                                    - kinesis:GetRecords
+                                    - kinesis:GetShardIterator
+                                    - kinesis:DescribeStream
+                                    - kinesis:ListStreams
+                                Resource: !Sub arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:stream/${InputStream}
+    JavaLambdaFunction:
+        Type: AWS::Lambda::Function
+        Properties:
+            Description: Java consumer
+            Runtime: java8
+            MemorySize: 512
+            Timeout: 90
+            Role: !GetAtt JavaLambdaRole.Arn
+            Handler: com.amazonaws.crypto.examples.kinesisdatakeycaching.LambdaDecryptAndWrite::handleRequest
```

Comment:
Kinesis with Lambda should be separate maven module.
Shouldn't add dependency on Kinesis to existing applications which don't use these services.

Reply:
Note that this is the example from the docs: http://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/sample-cache-example-java.html

Kinesis is shown as a motivating example of where you might want to use the caching feature.